Implement mobile responsive search for small screen sizes

maurizi opened the pull request:
With these changes, all advanced search functionality is now available on small screen sizes.

Map page on page load:
![screen shot 2016-10-10 at 14 15 46](https://cloud.githubusercontent.com/assets/4432106/19246521/bf07b702-8ef4-11e6-9270-6a54307616fc.png)

When tapping on the location search box:
![screen shot 2016-10-10 at 14 18 43](https://cloud.githubusercontent.com/assets/4432106/19246524/bf0c21fc-8ef4-11e6-955c-42ee6303200e.png)

After tapping on the "Advanced search" toggle in the header:
![screen shot 2016-10-10 at 14 19 59](https://cloud.githubusercontent.com/assets/4432106/19246520/bf07550a-8ef4-11e6-8d74-84edd8c6f65e.png)

Search buttons are sticky when scrolling:
![screen shot 2016-10-10 at 14 20 05](https://cloud.githubusercontent.com/assets/4432106/19246522/bf087458-8ef4-11e6-9777-e1f9c2ccf54c.png)

For medium-sized screens we now always show both the species and location search boxes (by hiding the map logo):
![screen shot 2016-10-10 at 14 20 24](https://cloud.githubusercontent.com/assets/4432106/19246523/bf0c131a-8ef4-11e6-82fe-820ea5e95ca4.png)

Connects to https://github.com/OpenTreeMap/otm-core/issues/1127

maurizi commented:
@lederer I implemented the changes you suggested in 2277e49

lederer commented:
@maurizi Thanks for that. I'd forgotten to update the PR with the mock yesterday.

lederer commented:
👍

RickMohr commented:
The little user image photo is gone. Before:
![image](https://cloud.githubusercontent.com/assets/2162993/19325143/c7c026e6-9092-11e6-9e57-6b5c16a9b635.png)
After:
![image](https://cloud.githubusercontent.com/assets/2162993/19325159/d5d82af8-9092-11e6-8cef-fafb04a8d503.png)

RickMohr commented:
If I do a location search on mobile and want to clear the results I have to think to click "Advanced Search", which might be a stretch for many. (Then after I click "Reset" I have to click the "X" to get back out.)

Likewise, if I do an advanced search and want to clear the results I have to click "Advanced Search" + Reset + X.

Can we find a way to get that "Reset" button onto the header for mobile?

RickMohr commented:
The black header bar is slightly taller in "mobile" mode than the other modes, and the first label is a little further to the right:
![image](https://cloud.githubusercontent.com/assets/2162993/19325979/9071e9b4-9096-11e6-8327-70cb31fe707f.png)  ![image](https://cloud.githubusercontent.com/assets/2162993/19325995/a41cfe22-9096-11e6-9597-598383c60ac8.png)
![image](https://cloud.githubusercontent.com/assets/2162993/19326003/af580304-9096-11e6-888f-f1b55c1c3eed.png)

(Watch it dynamically by changing the window width.)

RickMohr commented:
Can we put "Tree Care" at the bottom in mobile mode? Seems least important, but it's shown first.

RickMohr commented:
In mobile mode, the "Search by Species" and "Search by Location" headers should be the same color as the other headers:
![image](https://cloud.githubusercontent.com/assets/2162993/19326255/c39d3702-9097-11e6-8ec0-3602fdf03cb6.png)

RickMohr commented:
There's a temporary layout glitch when loading in non-mobile mode. In the "Search by Species" and "Search by Location" boxes the placeholder text overlays the hamburger icon for a noticeable amount of time, looking ugly. Then it's resolved.

Not present on staging.

RickMohr commented:
That's all I can find for now.

maurizi commented:
> There's a temporary layout glitch when loading in non-mobile mode. In the "Search by Species" and "Search by Location" boxes the placeholder text overlays the hamburger icon for a noticeable amount of time, looking ugly. Then it's resolved.

I saw this too, and I have no idea how to fix it.  @lederer do you have any ideas? I could use an assist.

maurizi commented:
@RickMohr Most of the comments you raised have been addressed, this is ready for another look.

lederer commented:
@maurizi Re that layout glitch, I haven't been able to track down the precise cause, but I've got some evidence that might help you get there:

1. Using Chrome Inspector, I recorded the timeline of the page reload.
1. The frame that gets rendered at 1.64s has the offending styles, which come from `input[type="text"]`.
1. That frame remains onscreen until 2.41s, or ~7.64ms, which is a lifetime.
1. The next frame is when the correct styles for those elements get rendered. They come from `.search-block-wrapper .search-block .search-field-group .twitter-typeahead input[type="text"]`.

So at the risk of a tautology, I reckon there's something going on with that huge rendering delay / low FPS, and with Twitter Typeahead. Holler and I can show you the timeline I captured.

maurizi commented:
@lederer That was a huge help :smile: . `twitter-typeahead` is a class that gets added by JS, so depending on it for styling stuff wasn't a good idea.

@RickMohr I put back the user logo and fixed a styling bug with it, fixed the species/location label coloring, moved tree care to the bottom of  the search area, and reworked how the advanced / reset buttons work.

thanks for all the thoughtful comments!

RickMohr commented:
@maurizi, all looks good!

:+1:
